src/main.rs

```diff
@@ -0,0 +1,188 @@
+use std::env;
+use std::fmt;
+use std::fs::{self,File};
+use std::io::{self,Read};
+use std::os::unix::fs as unix_fs;
+use std::path::{Path,PathBuf};
+use std::process::Command;
+
+struct PathError {
+    message: String,
+    path: PathBuf,
+}
+
+impl fmt::Display for PathError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}: {}", self.message, self.path.display())
+    }
+}
+
+// Creates a file if it does not exist
+fn ensure_file_exists<P: AsRef<Path>>(file_path: P, is_directory: Option<bool>) -> io::Result<()> {
```

Comment:
I'd probably split this into ensure_file_exists and ensure_directory_exists rather than using a bool or an Option<bool> here.